weave_query/weave_query/artifact_wandb.py

```diff
@@ -994,7 +994,7 @@ def uri_path(self) -> str:
 
     @property
     def resolved_artifact_uri(self) -> "WeaveWBArtifactURI":
-        if self.version and likely_commit_hash(self.version):
+        if self.version and (likely_commit_hash(self.version) or is_valid_version_index(self.version)):
```

Comment:
As per this comment, the `resolved_artifact_uri` just needs to be a specific, immutable version like v4 instead of an alias for idempotency of cache keys. The existing check enforced the version in the uri to be a commit hash which caused issues while fetching the artifact in a portfolio collection since we don't store the commit hashes for those memberships. 

I initially went about updating the `FetchArtifactByCommitHash` to allow fetching an artifact via its portfolio membership and sequence commit hash but this fix to use the version index instead makes more sense. (Pr here https://github.com/wandb/core/pull/27670)